src/rng/backends/rocrand/rocrand_helper.hpp

```diff
@@ -0,0 +1,194 @@
+ * such, the U.S. Government has been granted for itself and others acting on
+ * its behalf a paid-up, nonexclusive, irrevocable, worldwide license in the
+ * Software to reproduce, distribute copies to the public, prepare derivative
+ * works, and perform publicly and display publicly, and to permit others to do
+ * so.
+ ******************************************************************************/
+
+/**
+ * @file curand_helper.cpp : contains the implementation of all the routines
+ * for HIP backend
+ */
+#ifndef _MKL_RNG_ROCRAND_HELPER_HPP_
+#define _MKL_RNG_ROCRAND_HELPER_HPP_
+//#include <cuda.h>
+#include <rocrand.h>
+
+#include <complex>
+
+#include "oneapi/mkl/types.hpp"
+
+namespace oneapi {
+namespace mkl {
+namespace rng {
+namespace rocrand {
+
+class rocrand_error : virtual public std::runtime_error {
+protected:
+    inline const char* rocrand_error_map(rocrand_status error) {
+        switch (error) {
+            case ROCRAND_STATUS_SUCCESS: return "ROCRAND_STATUS_SUCCESS";
+
+            case ROCRAND_STATUS_NOT_CREATED: return "ROCRAND_STATUS_NOT_CREATED";
+
+            case ROCRAND_STATUS_ALLOCATION_FAILED: return "ROCRAND_STATUS_ALLOCATION_FAILED";
+
+            case ROCRAND_STATUS_TYPE_ERROR: return "ROCRAND_STATUS_TYPE_ERROR";
+
+            case ROCRAND_STATUS_OUT_OF_RANGE: return "ROCRAND_STATUS_OUT_OF_RANGE";
+
+            case ROCRAND_STATUS_LENGTH_NOT_MULTIPLE: return "ROCRAND_STATUS_LENGTH_NOT_MULTIPLE";
+
+            case ROCRAND_STATUS_DOUBLE_PRECISION_REQUIRED:
+                return "ROCRAND_STATUS_DOUBLE_PRECISION_REQUIRED";
+
+            case ROCRAND_STATUS_LAUNCH_FAILURE: return "ROCRAND_STATUS_LAUNCH_FAILURE";
+
+            case ROCRAND_STATUS_VERSION_MISMATCH: return "ROCRAND_STATUS_VERSION_MISMATCH";
+
+            case ROCRAND_STATUS_INTERNAL_ERROR: return "ROCRAND_STATUS_INTERNAL_ERROR";
+
+            default: return "<unknown>";
+        }
+    }
+
+    int error_number; ///< Error number
+public:
+    /** Constructor (C++ STL string, rocrand_status).
+   *  @param msg The error message
+   *  @param err_num error number
+   */
+    explicit rocrand_error(std::string message, rocrand_status result)
+            : std::runtime_error((message + std::string(rocrand_error_map(result)))) {
+        error_number = static_cast<int>(result);
+    }
+
+    /** Destructor.
+   *  Virtual to allow for subclassing.
+   */
+    virtual ~rocrand_error() throw() {}
+
+    /** Returns error number.
+   *  @return #error_number
+   */
+    virtual int getErrorNumber() const throw() {
+        return error_number;
+    }
+};
+
+class cuda_error : virtual public std::runtime_error {
+protected:
+    inline const char* cuda_error_map(hipError_t result) {
+        switch (result) {
+            case HIP_SUCCESS: return "HIP_SUCCESS";
+
+            case HIP_ERROR_NOT_INITIALIZED: return "HIP_ERROR_NOT_INITIALIZED";
+
+            case hipErrorInvalidContext: return "HIP_ERROR_INVALID_CONTEXT";
```

Comment:
lower case or upper case?

Reply:
yes this is wrong here I use the error codes from here: https://rocm-developer-tools.github.io/HIP/group__GlobalDefs.html#gadf5010f6e140a53ecbdf949e73e87594